sdk/eventhub/azure-eventhub-checkpointstoretable/tests/test_storage_table_partition_manager.py

```diff
@@ -1,8 +1,119 @@
 import pytest
 from azure.eventhub.extensions.checkpointstoretable import TableCheckpointStore
+from azure.data.tables import TableServiceClient
+import uuid
+import warnings
+import random
+import os
 
+STORAGE_CONN_STR = [os.environ.get("AZURE_STORAGE_CONN_STR", "Azure Storage Connection String")]
 
-def test_constructor():
-    client = TableCheckpointStore()
-    assert client is not None
-    
+def get_live_storage_table_client(storage_connection_str):
+    try:
+        table_name = chr(random.randint(ord('a'), ord('z'))) +(uuid.uuid4().hex)
+        table_service_client = TableServiceClient.from_connection_string(storage_connection_str)
+        table_service_client.create_table_if_not_exists(table_name)
+        return storage_connection_str, table_name
+    except:
+        pytest.skip("Storage table client can't be created")
+
+def remove_live_storage_table_client(storage_connection_str, table_name):
+    try:
+        table_service_client = TableServiceClient.from_connection_string(storage_connection_str)
+        table_service_client.delete_table(table_name)
+    except:
+        warnings.warn(UserWarning("storage table teardown failed"))
+
+def _claim_and_list_ownership(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
+    ownership_cnt = 8
+
+    checkpoint_store = TableCheckpointStore.from_connection_string(
+        storage_connection_str, table_name)
+    ownership_list = checkpoint_store.list_ownership(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(ownership_list) == 0
+
+    ownership_list = []
+
+    for i in range(ownership_cnt):
+            ownership = {}
+            ownership['fullyqualifiednamespace'] = fully_qualified_namespace
+            ownership['eventhubname'] = eventhub_name
+            ownership['consumergroup'] = consumer_group
+            ownership['ownerid'] = 'ownerid'
+            ownership['partitionid'] = str(i)
+            ownership['etag'] = None
+            ownership_list.append(ownership)
+    checkpoint_store.claim_ownership(ownership_list)
+
+    ownership_list = checkpoint_store.list_ownership(fully_qualified_namespace,eventhub_name,consumer_group)
```

Comment:
Just to make sure we've persisted the ownership like we think we did, we can compare this list to the output of `claim_ownership`. The values should match exactly (you may need to sort them to get them to exactly match).

Reply:
sounds good

Reply:
while making the changes I see that the metadata for type of last_modified_time is different. when you call list_checkpoint it query entities and these entites metadata are etag and timestamp while when you create an entity the metadata are etag and date. timestamp and date have two different types in which date type is date.date while timestamp type is not. And last_modified_time is supposed to have date.date type. To solve this issue I was thinking of having a field in ownership for last_modified_time that has type date.date.

Reply:
Can you convert the timestamp into a date?

Reply:
here's an example of a function that convert a date to a timestamp: https://github.com/Azure/azure-sdk-for-python/blob/c8291ac6cb0dbd865da03a88dd2bcb9279e2c4a6/sdk/eventhub/azure-eventhub-checkpointstoreblob/azure/eventhub/extensions/checkpointstoreblob/_blobstoragecs.py#L28

Reply:
Yes thank swathi for the reference. I was looking into that method yesterday but noticed that  last_modified_time type it says that it must be date.date type.

Reply:
If you want to convert timestamp -> dt, this line should be useful: https://github.com/Azure/azure-sdk-for-python/blob/c8291ac6cb0dbd865da03a88dd2bcb9279e2c4a6/sdk/eventhub/azure-eventhub-checkpointstoreblob/azure/eventhub/extensions/checkpointstoreblob/_blobstoragecs.py#L24